### PR TITLE
Updates for JS bug bash feedback

### PR DESCRIPTION
**Pull request by navyasric**
- Made http and port consistent on Redirect URIs
- Added details for downloading and extracting project to short dir name
- Added step to run the project
- Updated Library version in reference
- Added details on authority, scopes in more info.

### CODE REVIEW

**Comment by navyasric** [comment]
@andretms @jmprieur Please review the updates in doc content.

**Comment by andretms** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/nc_spa_quickstart/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C9%5C14%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201809140237287383-12%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=Yw9no9GO6NIDH4seRQ786D15PBdOW1DjcAnqjrYJVbs%3D&st=2018-09-14T02%3A33%3A31Z&se=2018-10-15T02%3A38%3A31Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 1]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
 
 > [!div renderon="docs"]
 > ## Register your application and download your quickstart app
-> You have two options to start your quickstart application:
-> * [Express] [Option 1: Register and auto configure your app and then download your code sample](#option-1-register-and-auto-configure-your-app-and-then-download-your-code-sample)
-> * [Manual] [Option 2: Register and manually configure your application and code sample](#option-2-register-and-manually-configure-your-application-and-code-sample)
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:33]: This need to be: #option-2-register-and-manually-configure-your-app-and-then-download-the-code-sample

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 2]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
 
 > [!div renderon="docs"]
 > ## Register your application and download your quickstart app
-> You have two options to start your quickstart application:
-> * [Express] [Option 1: Register and auto configure your app and then download your code sample](#option-1-register-and-auto-configure-your-app-and-then-download-your-code-sample)
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:32]: This need to be: #option-1-register-and-auto-configure-your-app-and-then-download-the-code-sample

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 3]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
...
-> ### Option 2: Register and manually configure your application and code sample
+> ### Option 2: Register and manually configure your app and then download the code sample
 > #### Step 1: Register your application
-> 
+>
 > 1. To register an application, go to the [Azure portal - Application Registration](https://aka.ms/registeredappsprod) and select **New registration**.
-> 1. Enter a name for your application, add `http://localhost:44316/` in **Reply URL**, and click **Register**.
-> 1. Select **Authentication** menu, set **ID tokens** under **Implicit Grant**, and then select **Save**.
+> 1. Enter a name for your application.
+> 1. Select **Accounts in any Azure AD directory and personal Microsoft accounts** under Supported Microsoft accounts.
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: I would suggest: 
>1. Select **Accounts in any Azure AD directory and personal Microsoft accounts** under *Supported Microsoft accounts* (Optional)

**Review by jmprieur — APPROVED** [review]
@navyasric : this looks very good !

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 4]
```diff
@@ -74,12 +81,20 @@ var msalconfig = {
 };
 ```
 > [!NOTE]
-> If you use [Visual Studio 2017](https://visualstudio.microsoft.com/downloads/), the Redirect URL will be set to to *http:/<span/>/localhost:44316/* as it is configured in the code sample's project. If you use [Node.js](https://nodejs.org/en/download/) or any other web server, set the redirect URL to `http://localhost:44316/` and then configure the server to start listening this port. 
+> If you use [Visual Studio 2017](https://visualstudio.microsoft.com/downloads/), the redirect URI will be set to to *http:/<span/>/localhost:30662/* as it is configured in the code sample's project. If you use [Node.js](https://nodejs.org/en/download/) or any other web server, set the redirect URI to *http:/<span/>/localhost:30662/* and then configure the server to start listening on this port.
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: replace *http:/<span/>/localhost:30662/* with `http://localhost:30662/` for consistance

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 5]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
...
 
-#### Step 2: Download your web server or project
+#### Step 2: Download the project
 
-* [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
+You can choose one of these options suitable to your development environment.
 * [Download the core project files - for a web server, such as Node.js](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/core.zip)
+* [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
+
+Extract the zip file of the project to a local directory such as C://spa.
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: For consistence with other quickstarts, I would keep like:

Extract the zip file to a local folder (for example, **C:\Azure-Samples**)

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 6]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
...
 
-#### Step 3: Configure your JavaScript SPA
+#### Step 3: Configure your JavaScript app
 
 > [!div renderon="docs"]
-> Edit `msalconfig.js` and replace `Enter_the_Application_Id_here` with the Application ID your application:
+> Edit `msalconfig.js` and replace `Enter_the_Application_Id_here` with the Application ID of your application:
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: > Edit `msalconfig.js` and replace `Enter_the_Application_Id_here` with the Application ID of the application you just registered:
>> [!TIP]
>> To find the *Application ID*, go to *Overview* page.

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md (hunk 4 above)
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: Replace *http:/<span/>/localhost:30662/* with `http://localhost:30662/`

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 7]
```diff
@@ -74,12 +81,20 @@ var msalconfig = {
...
+#### Step 4: Run the project
+
+If using Visual Studio, press **F5** to run your project.
+
+If using Node.js, on a command line you can run the following from the project's directory to start the server:
+ ```batch
+ npm install
+ node server.js
+ ```
+In a browser window navigate to the `http://localhost:30662/` location and you see the **Call Microsoft Graph API** button to start sign-in.
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: Suggestion: 
Open the web browser and navigate to `http://localhost:30662/`. Click **Call Microsoft Graph API** button to start sign-in.

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 8]
```diff
@@ -110,21 +125,21 @@ var userAgentApplication = new Msal.UserAgentApplication(msalconfig.clientID, nu
 > |Where  |  |
 > |---------|---------|
 > |`ClientId`     |Application Id from the application registered in the Azure Portal|
-> |`authority`    |It is the authority URL. *null* sets the default authority to <i>https:<span/>//login.microsoftonline.com/common</i>|
-> |`loginCallBack`| Name of a callback method called after the authentication|
-> |`redirectUri`  |URL where users are sent after authentication against Azure AD v2 Endpoint|
+> |`authority`    |It is the authority URL. Passing *null* sets the default authority to <i>https:<span/>//login.microsoftonline.com/common</i>. If using a specific tenant, set authority to <i>https:<span/>//login.microsoftonline.com/&lt;tenant name or ID&gt;</i>|
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: Suggestion: 
It is the authority URL. Passing *null* sets the default authority to `https://login.microsoftonline.com/common`. If your app is single-tenant (targeting accounts in one directory only), set this value to `https://login.microsoftonline.com/<tenant name or ID>`

**Review by andretms — CHANGES_REQUESTED** [review]

**Comment by andretms** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md](https://github.com/andretms/quickstarts-test/blob/nc_spa_quickstart/articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C9%5C14%5C58599950-09d3-b0f6-1b9c-85cb52a0ce7a%5CPullRequest%5C201809141929319754-12%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=nNKRYjBcIZdVndfI6KSWvnXF%2FpR%2B8v65raKDoHUpruA%3D&st=2018-09-14T19%3A25%3A35Z&se=2018-10-15T19%3A30%3A35Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md (hunk 1 above)
Comment (navyasric) [reply articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md:33]: @andretms Thanks for comments. Added updates, please check  to approve.

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 9]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
 
 > [!div renderon="docs"]
 > ## Register your application and download your quickstart app
-> You have two options to start your quickstart application:
-> * [Express] [Option 1: Register and auto configure your app and then download your code sample](#option-1-register-and-auto-configure-your-app-and-then-download-your-code-sample)
-> * [Manual] [Option 2: Register and manually configure your application and code sample](#option-2-register-and-manually-configure-your-application-and-code-sample)
+> You have two options to start your quickstart app:
+> * [Express] [Option 1: Register and auto configure your app and then download the code sample](#option-1-register-and-auto-configure-your-app-and-then-download-your-code-sample)
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: This is still incorrect - have to match the line 35: #option-1-register-and-auto-configure-your-app-and-then-download-the-code-sample

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md [hunk 10]
```diff
@@ -28,41 +28,48 @@ In this quickstart, you'll learn how to use a code sample that demonstrates how
...
+> * [Manual] [Option 2: Register and manually configure your app and then download the code sample](#option-2-register-and-manually-configure-your-application-and-code-sample)
 >
```
Comment (andretms) [inline articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: This need to match the line 43:
#option-2-register-and-manually-configure-your-app-and-then-download-the-code-sample

File: articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md (hunk 3 above)
Comment (andretms) [reply articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md]: I would still add the italic or bold under Supported Microsoft Accounts

**Review by andretms — APPROVED** [review]

**Pull request merged**